autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/ExpandedCourseCard.tsx

```diff
@@ -110,6 +143,8 @@ const ExpandedCourseCard: React.FC<ExpandedCourseCardProps> = ({
             }, term));
           }}
           onInputChange={(evt: object, val: string, reason: string): void => {
+            val = remove_all_spaces(val);
```

Comment:
See function comments. As remove duplicate spaces, this would ideally be called in the add_space_between_course_subject_and_number function before additional string processing.